src/main/java/com/google/sps/data/Journey.java

```diff
@@ -16,31 +16,32 @@
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Date;
-import java.util.Map;
+import java.util.List;
 import java.lang.reflect.Array;
 import com.google.maps.errors.ApiException;
 import java.io.IOException;
 import java.lang.InterruptedException;
-
-import java.util.HashMap;
+import java.time.Duration;
+import java.time.Instant;
 
 /**
  * Class that represents a delivery journey.
  */
 public class Journey {
   // The number of seconds the journey lasts.
   private int minTime;
-  // Array of pairs (a, b) meaning that waypoint with value a in waypoints must be visited before
-  // waypoint b.
+  // Array of pairs (a, b) meaning that waypoint with index a in waypoints must be visited before
+  // waypoint with index b.
   private ArrayList<Pair> restrictions;
   // The starting point for the journey.
   private Point start;
-  // Hashmap with keys representing Point objects and values representing their index. E.g. when
-  // a new Point is added to the hashmap, its value will be the equal to the number of elements.
-  // The hashmap is preferred to an ArrayList so that Point objects can be removed efficiently.
-  private HashMap<Point, Integer> waypoints;
-  // TODO[ak47na]: add PathFinder object, interface and the classes that implement it: 
-  // GoogleMapsPathFinder used for production and ManhattanDistancePathFinder for testing.
+  // Array of Point objects representing the waypoints that must be visited not including the start
+  // point. The index of each waypoint in the array will be used to mark order restrictions.
+  private List<Point> waypoints;
+  // Array of encoded string representations of the keys of orders from Datastore assigned to the journey.
+  private List<String> orderKeys;
```

Comment:
I am a bit confused as to how this will work. Don't you need to store the orders for each waypoint?

Otherwise, how will the courier know what to pick up from each library and what to deliver at each destination? A user might have multiple open orders assigned to multiple couriers, no?